actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserSelfDeclarationManagementActor.java

```diff
@@ -114,6 +127,27 @@ private void updateUserDeclarations(Request actorMessage) {
         userMap, targetObject, correlatedObject, actorMessage.getContext());
   }
 
+  private Map<String, Object> updateSchoolInfoInSelfDeclaration(
+      String userId, RequestContext context, Map<String, Object> userInfo) {
+    UserOrgDao userOrgDao = UserOrgDaoImpl.getInstance();
+    Response res = userOrgDao.getUserOrgDetails(userId, null, context);
+    List<Map<String, Object>> userOrgLst = (List<Map<String, Object>>) res.get(JsonKey.RESPONSE);
+    if (!(userOrgLst.size() == 1
+        && userOrgLst.get(0).get(JsonKey.ORGANISATION_ID).equals(custodianRootOrgId))) {
+      for (Map<String, Object> stringObjectMap : userOrgLst) {
+        if (!stringObjectMap.get(JsonKey.ORGANISATION_ID).equals(custodianRootOrgId)) {
+          String organisationId = (String) stringObjectMap.get(JsonKey.ORGANISATION_ID);
+          Organisation organisation = organisationClient.esGetOrgById(organisationId, context);
+          if (null != organisation) {
+            userInfo.put(JsonKey.DECLARED_SCHOOL_UDISE_CODE, organisation.getId());
```

Comment:
externalid should be used instead of org id

Reply:
Done